### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by konsumlamm**
Fix some grammar mistakes, use code blocks and add some links.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,7 +17,7 @@ Icon?
 # Haskell #
 ###########
 
-dist
+dist-newstyle
```
Comment (Bodigrim) [inline .gitignore:20]: Please retain dist as well.

File: src/Data/Bit/F2Poly.hs [hunk 2]
```diff
@@ -53,20 +53,20 @@ import GHC.Integer.Logarithms
 #endif
 
 -- | Binary polynomials of one variable, backed
--- by an unboxed 'Data.Vector.Unboxed.Vector' 'Bit'.
+-- by an unboxed @'Data.Vector.Unboxed.Vector' 'Bit'@.
```
Comment (Bodigrim) [inline src/Data/Bit/F2Poly.hs:56]: Please revert this and similar.

File: src/Data/Bit/F2Poly.hs [hunk 3]
```diff
@@ -320,7 +320,7 @@ toBigNat (ByteArray arr) = BN# arr
 
 -- | Execute the extended Euclidean algorithm.
 -- For polynomials @a@ and @b@, compute their unique greatest common divisor @g@
--- and the unique coefficient polynomial @s@ satisfying @as + bt = g@.
+-- and the unique coefficient polynomial @s@ satisfying @a * s + b * t = g@.
```
Comment (Bodigrim) [inline src/Data/Bit/F2Poly.hs:323]: That's probably a good opportunity to use LaTeX.

File: src/Data/Bit/Immutable.hs [hunk 4]
```diff
@@ -189,10 +189,10 @@ cloneToWords v = runST $ do
   U.unsafeFreeze w
 {-# INLINE cloneToWords #-}
 
--- | Cast a unboxed vector of 'Word8'
+-- | Cast an unboxed vector of 'Word8'
 -- to an unboxed vector of bits.
 --
--- On big-endian architectures 'castFromWords8'
+-- On big-endian architectures 'castFromWords8', this
```
Comment (Bodigrim) [inline src/Data/Bit/Immutable.hs:195]: I don't understand this change, it does not make sense to me.

**Review by Bodigrim — COMMENTED** [review]
Thanks, I appreciate this!

File: src/Data/Bit/Immutable.hs (hunk 4 above)
Comment (konsumlamm) [reply src/Data/Bit/Immutable.hs:195]: The "this" refers to the function itself (`castFromWords8`), but I can change it back if you find that easier to read.

File: src/Data/Bit/F2Poly.hs (hunk 3 above)
Comment (konsumlamm) [reply src/Data/Bit/F2Poly.hs]: I replaced this with `a \cdot s + b \cdot t = g`, is that good?

File: src/Data/Bit/Internal.hs [hunk 5]
```diff
@@ -53,27 +53,27 @@ import GHC.Exts
...
 -- but random writes are up to 20% slower.
 newtype Bit = Bit { unBit :: Bool }
   deriving (Bounded, Enum, Eq, Ord, FiniteBits, Bits, Typeable, Generic, NFData)
 #endif
 
 -- | There is only one lawful 'Num' instance possible
--- with '+' = 'xor' and
--- 'fromInteger' = 'Bit' . 'odd'.
+-- with @'+' = 'xor'@ and
+-- @'fromInteger' = 'Bit' . 'odd'@.
```
Comment (Bodigrim) [inline src/Data/Bit/Internal.hs:76]: Please revert this change.

File: src/Data/Bit/Internal.hs [hunk 6]
```diff
@@ -408,8 +408,8 @@ instance MV.MVector U.MVector Bit where
 #ifndef BITVEC_THREADSAFE
 
 -- | Flip the bit at the given position.
--- No bounds checks are performed.
--- Equivalent to 'flip' 'Data.Vector.Unboxed.Mutable.unsafeModify' 'Data.Bits.complement',
+-- No bound checks are performed.
+-- Equivalent to @'flip' 'Data.Vector.Unboxed.Mutable.unsafeModify' 'Data.Bits.complement'@,
```
Comment (Bodigrim) [inline src/Data/Bit/Internal.hs:412]: Please revert @…@ here and below.

**Review by Bodigrim — APPROVED** [review]

**Comment by Bodigrim** [comment]
Thanks, merged!
